src/GraphQL/Introspection/ISchemaFilter.cs

```diff
@@ -0,0 +1,41 @@
+using GraphQL.Types;
+
+namespace GraphQL.Introspection
+{
+    public interface ISchemaFilter
+    {
+        bool Type(IGraphType type);
+        bool Field(IGraphType parent, IFieldType field);
+        bool Argument(IFieldType field, QueryArgument argument);
+        bool EnumValue(EnumerationGraphType parent, EnumValueDefinition enumValue);
+        bool Directive(DirectiveGraphType directive);
+    }
+
+    public class DefaultSchemaFilter : ISchemaFilter
```

Comment:
If you intend to use this "hide feature", then you will probably need to define only a part of the behavior, and reuse the rest, so it can be convenient to do all those methods `virtual`.